### PR TITLE
chore: depreacte experiments.parallelCodeSplitting

### PR DESCRIPTION
**Pull request by JSerFeng**
## Summary

This options is depreacted, it has huge performance regression in some edge cases where the chunk graph has lots of cycles, we'll improve the performance of build_chunk_graph in the future instead.

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: website/docs/en/config/experiments.mdx [hunk 1]
```diff
@@ -150,15 +150,15 @@ Starting from v1.4.0, Rspack enables incremental builds for all phases by defaul
 
 ## experiments.parallelCodeSplitting
 
-<ApiMeta addedVersion="1.2.0" />
+<ApiMeta deprecatedVersion="1.6.2" />
```
Comment (chenjiahan) [inline website/docs/en/config/experiments.mdx]: We can move the deprecated options to the bottom of the page

Comment (chenjiahan) [reply website/docs/en/config/experiments.mdx]: See: https://rspack.rs/config/experiments#experimentstoplevelawait

Comment (JSerFeng) [reply website/docs/en/config/experiments.mdx]: Done
